currying/README.md

```diff
@@ -0,0 +1,34 @@
+# Currying
+
+Currying is the concept of taking a single function that takes multiple arguments into a chained sequence
+of functions that each take a single function.
+
+Sample Code:
+
+> As you can see this function accepts two integers and returns one
+this is not a curried function, nor is it a high order function.
+
+```
+add :: (Integer, Integer) -> Integer
+```
+
+> This function takes one integer, followed by a function that takes an 
+integer, and finall returns an integer, this is an example of currying and 
+also a high order function.
+
+```
+add :: Integer -> Integer -> Integer
+```
+
+Currying is often confused with partial-evaluation, however the key difference is with currying you 
+are not infact creating a function accepts less than the required number of arguments, you are actually
+creating a chain of functions that each accept one function.
```

Comment:
Or at least, creating a generator that can produce functions of this description.